src/call.cpp

```diff
@@ -923,33 +923,72 @@ char * call::get_last_header(const char * name)
     if (name[len - 1] == ':') {
         return get_header(last_recv_msg, name, false);
     } else {
+
         char with_colon[MAX_HEADER_LEN];
-        sprintf(with_colon, "%s:", name);
-        return get_header(last_recv_msg, with_colon, false);
+        const char *sep = strrchr(name, ':');
+
+        if (!sep){
+            sprintf(with_colon, "%s:", name);
+            return get_header(last_recv_msg, with_colon, false);
+        } else {
+            if (!strcmp(sep, ":value")) {
+                snprintf(with_colon, len - 4, "%s", name);
+                return get_header(last_recv_msg, with_colon, true);
+            }
+            if (!strcmp(sep, ":uri")) {
+                snprintf(with_colon, len - 2, "%s", name);
+                char * last_header_uri = get_last_request_uri(with_colon);
+                char * ret;
+                if (!(ret = (char *)malloc(strlen(last_header_uri + 1)))){
```

Comment:
get_header() returns a static buffer (yuck, I know).

You can't start returning malloc'ed memory all of a sudden: memory leaks.